src/index.js

```diff
@@ -520,6 +553,206 @@ class EventsCoreAPI {
+   * 
+   * @param {*} pubKeyUrl cloud front url of format https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+   * @param {*} state aio-lib-state client
+   * @returns public key
+   */
+  async __fetchFromCacheOrApi (pubKeyUrl, state) {
+    let publicKey, publicKeyFileName
+    try {
+      // public key url is the cloud front url in this format https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+      publicKeyFileName = pubKeyUrl.substring(pubKeyUrl.lastIndexOf('/') + 1)
+      const res = await state.get(publicKeyFileName)
+      publicKey = res.value
+      logger.info('public key %s fetched from aio lib state', publicKey)
+    } catch (error) {
+      logger.error('error fetching from aio lib state due to => %s', error.message)
+      throw error
+    }
+
+    if (publicKey) {
+      return publicKey
+    }
+
+    logger.info('public key for url %s not present in state lib cache, fetching from api..', pubKeyUrl)
+    // fetch from api and set in cache default expiry 24h
+    publicKey = this.__fetchPublicKey(pubKeyUrl)
+    await state.put(publicKeyFileName, { anObject: publicKey })
+    logger.info('fetched key set in the aio lib state')
+    return publicKey
+  }
+
+  /**
+   * Fetches public key using the cloud front public key url
+   * 
+   * @param {*} publicKeyUrl - cloud front public key url of format 
+   * https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+   * @returns public key
+   */
+  __fetchPublicKey (publicKeyUrl) {
+    let pubKey
+    fetch(publicKeyUrl)
+      .then(res => res.text())
+      .then(text => {
+        logger.debug('successfully fetched the public key %s from api', text)
+        pubKey = text
+      })
+      .catch(err => {
+        logger.error('error fetching the public key from api due to => %s', err.message)
+        throw err
+      })
+    return pubKey
+  }
+
+  /**
+   * Digital Signature Verification Helper
+   * 
+   * @param {*} digitalSignature1 - I/O Events generated digital signature1
+   * @param {*} digitalSignature2 - I/O Events generated digital signature2
+   * @param {*} signedPayload - I/O Events proper signed payload
+   * @param {*} pubKey1Pem - I/O Events generated PEM encoded public key1
+   * @param {*} pubKey2Pem - I/O Events generated PEM encoded public key1
+   * @param {*} recipientClientId - target recipient client id
+   * @returns {boolean} true if either signatures are valid or false
+   */
+  __verifySignature (digitalSignature1, digitalSignature2, signedPayload, pubKey1Pem, pubKey2Pem, recipientClientId) {
+    let result1, result2, publicKey1, publicKey2
+    try {
+      publicKey1 = this.__createCryptoPublicKey(pubKey1Pem)
+      result1 = this.__cryptoVerify(digitalSignature1, publicKey1, signedPayload)
+      if (result1) {
+        return result1
+      } 
+
+      publicKey2 = this.__createCryptoPublicKey(pubKey2Pem)
+      result2 = this.__cryptoVerify(digitalSignature2, publicKey2, signedPayload)
+      return result2
```

Comment:
Instead of duplicating the code and writing it like this, can we instead use a simple loop here?

Reply:
done